### PR TITLE
ROC-5387 save directions questionnaire

### PR DESCRIPTION
**Pull request by karthikkannankainos**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/ROC-5387

### Change description ###
Save Directions Questionnaire

### CODE REVIEW

File: definition/data/sheets/ComplexTypes.json [hunk 1]
```diff
@@ -128,6 +128,25 @@
   {"ID": "Party", "ListElementCode": "businessName", "FieldType": "Text", "ElementLabel": "Business Name", "SecurityClassification": "Public"},
   {"ID": "Party", "ListElementCode": "contactPerson", "FieldType": "Text", "ElementLabel": "Contact person", "SecurityClassification": "Public"},
   {"ID": "Party", "ListElementCode": "companiesHouseNumber", "FieldType": "Text", "ElementLabel": "Companies house number", "SecurityClassification": "Public"},
+  {"ID": "UnavailableDate", "ListElementCode": "unavailableDate", "FieldType": "Date", "ElementLabel": "Upload of defendant pin letter successful", "SecurityClassification": "Public"},
+  {"ID": "ExpertReportsRows", "ListElementCode": "expertReportDate", "FieldType": "Date", "ElementLabel": "Upload of defendant pin letter successful", "SecurityClassification": "Public"},
```
Comment (sabahirfan) [inline definition/data/sheets/ComplexTypes.json]: ElementLabel needs to be corrected for above two roww

File: definition/data/sheets/ComplexTypes.json [hunk 2]
```diff
@@ -128,6 +128,25 @@
...
+  {"ID": "ExpertReportsRows", "ListElementCode": "expertName", "FieldType": "Text", "ElementLabel": "Upload of defendant pin letter successful", "SecurityClassification": "Public"},
```
Comment (sabahirfan) [inline definition/data/sheets/ComplexTypes.json]: here as well

Comment (KiranVarma02) [reply definition/data/sheets/ComplexTypes.json]: done

File: definition/data/sheets/ComplexTypes.json (hunk 1 above)
Comment (KiranVarma02) [reply definition/data/sheets/ComplexTypes.json]: done

File: definition/data/sheets/ComplexTypes.json [hunk 3]
```diff
@@ -128,6 +129,24 @@
...
+  {"ID": "DirectionsQuestionnaire", "ListElementCode": "hearingLocationOption", "FieldType": "Text", "ElementLabel": "Hearing location option selected by Party", "SecurityClassification": "Public"},
+  {"ID": "DirectionsQuestionnaire", "ListElementCode": "hearingCourtAddress", "FieldType": "Text", "ElementLabel": "Court hearing location address by Party", "SecurityClassification": "Public"},
+  {"ID": "DirectionsQuestionnaire", "ListElementCode": "exceptionalCircumstancesReason", "FieldType": "Text", "ElementLabel": "Reason for exceptional circumstances", "SecurityClassification": "Public"},
+  {"ID": "DirectionsQuestionnaire", "ListElementCode": "unavailableDates", "FieldType": "Collection", "FieldTypeParameter": "Date","ElementLabel": "Dates unavailable to attend hearing", "SecurityClassification": "Public"},
+  {"ID": "DirectionsQuestionnaire", "ListElementCode": "languageInterpreted", "FieldType": "Text", "ElementLabel": "Language interpretation requirement", "SecurityClassification": "Public"},
```
Comment (sabahirfan) [inline definition/data/sheets/ComplexTypes.json]: Minor: prefer to name it `languageInterpreter`

File: definition/data/sheets/ComplexTypes.json [hunk 4]
```diff
@@ -128,6 +129,24 @@
...
+  {"ID": "DirectionsQuestionnaire", "ListElementCode": "signLanguageInterpreted", "FieldType": "Text", "ElementLabel": "Sign language interpretation requirement", "SecurityClassification": "Public"},
```
Comment (sabahirfan) [inline definition/data/sheets/ComplexTypes.json]: Same here

File: definition/data/sheets/ComplexTypes.json (hunk 3 above)
Comment (karthikkannankainos) [reply definition/data/sheets/ComplexTypes.json]: interpreter seems person who does. is it okay to have a languageInterpretation with value as YES or NO would be better? your thoughts?

Comment (sabahirfan) [reply definition/data/sheets/ComplexTypes.json]: The value is text and not Yes/No

Comment (karthikkannankainos) [reply definition/data/sheets/ComplexTypes.json]: then languageInterpreted make sense here. IMO

File: definition/data/sheets/AuthorisationCaseEvent.json [hunk 5]
```diff
@@ -35,6 +35,8 @@
   {"CaseTypeID": "MoneyClaimCase", "CaseEventID": "SupportUpdate", "UserRole": "caseworker-cmc", "CRUD": "C"},
   {"CaseTypeID": "MoneyClaimCase", "CaseEventID": "PinGenerationOperations", "UserRole": "caseworker-cmc", "CRUD": "CR"},
   {"CaseTypeID": "MoneyClaimCase", "CaseEventID": "SendingClaimantNotification", "UserRole": "caseworker-cmc", "CRUD": "CR"},
+  {"CaseTypeID": "MoneyClaimCase", "CaseEventID": "AssigningForDirections", "UserRole": "caseworker-cmc", "CRUD": "CR"},
```
Comment (simad) [inline definition/data/sheets/AuthorisationCaseEvent.json]: not sure the caseworker should have `C` permissions for these, might want to confirm with @SanchitaMoj

File: definition/data/sheets/ComplexTypes.json [hunk 6]
```diff
@@ -189,6 +208,7 @@
   {"ID": "Respondent", "ListElementCode": "claimantProvidedRepresentativeOrganisationPhone", "FieldType": "Text", "ElementLabel": "Organisation Phone", "SecurityClassification": "Public"},
   {"ID": "Respondent", "ListElementCode": "claimantProvidedRepresentativeOrganisationDxAddress", "FieldType": "Text", "ElementLabel": "Organisation DxAddress", "SecurityClassification": "Public"},
   {"ID": "Respondent", "ListElementCode": "claimantProvidedRepresentativeOrganisationEmail", "FieldType": "Email", "ElementLabel": "Organisation Email", "SecurityClassification": "Public"},
+  {"ID": "Respondent", "ListElementCode": "directionsQuestionnaire", "FieldType": "DirectionsQuestionnaire", "ElementLabel": "Respondent Directions Questionnaire", "SecurityClassification": "Public"},
```
Comment (simad) [inline definition/data/sheets/ComplexTypes.json]: shouldn't we also add the DQ object to the applicant?

Comment (sabahirfan) [reply definition/data/sheets/ComplexTypes.json]: For applicant it is in `ClaimantResponse`

Comment (simad) [reply definition/data/sheets/ComplexTypes.json]: 👍 I see

**Pull request merged**